internal/pkg/iostreams/io.go

```diff
@@ -156,6 +156,13 @@ func (s *system) ReadSecret() ([]byte, error) {
 		return nil, fmt.Errorf("prompting is disabled")
 	}
 
+	go func() {
```

Comment:
I tested this and unfortunately it breaks the terminal state. You can't use up/down errors to go through history, command isn't shown, etc.

Through some spelunking, you need to call RestoreState: https://github.com/golang/go/issues/31180#issuecomment-1133854964

I did run across this implementation which works really well: https://github.com/grisu48/common/blob/main/pkg/password/password_supported.go#L17

I think we should bring that implementation to this package. I like that it returns the error to the caller as well

Reply:
Great find as this was hidden when using Zsh.
Have imported state handling as suggested. 🙇🏻